examples/ltc_backend/CMakeLists.txt

```diff
@@ -0,0 +1,94 @@
+endif()
+message(STATUS "TORCH_DIR=${TORCH_DIR}")
+
+if ("${TORCH_MLIR_DIR}" STREQUAL "")
+    get_filename_component(TORCH_MLIR_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../" ABSOLUTE)
+endif()
+message(STATUS "TORCH_MLIR_DIR=${TORCH_MLIR_DIR}")
+
+###########################################################################
+# Setup PyTorch
+###########################################################################
+
+list(APPEND CMAKE_MODULE_PATH "${TORCH_MLIR_DIR}/python/torch_mlir/cmake/modules")
+include(TorchMLIRPyTorch)
+TorchMLIRProbeForPyTorchInstall()
+find_package(Torch 1.11 REQUIRED)
+
+TorchMLIRConfigurePyTorch()
+
+###########################################################################
+# Setup Python development
+###########################################################################
+
+list(APPEND CMAKE_MODULE_PATH "${TORCH_MLIR_DIR}/external/llvm-project/mlir/cmake/modules")
+include(MLIRDetectPythonEnv)
+mlir_configure_python_dev_packages()
+
+###########################################################################
+# Library definition
+###########################################################################
+
+include_directories(BEFORE
+        ${TORCH_INCLUDE_DIRS}
+        ${CMAKE_CURRENT_SOURCE_DIR}
+        ${CMAKE_CURRENT_BINARY_DIR}
+        ${Python3_INCLUDE_DIRS}
+        ${PYTHON_H_DIR}
+        ${TORCH_MLIR_DIR}/python
+        )
+link_directories("${TORCH_INSTALL_PREFIX}/lib")
+link_directories(${CMAKE_CURRENT_SOURCE_DIR}/ltc_backend/lib)
+add_link_options(-Wl,-rpath,$ORIGIN/ltc_backend/lib)
+
+file(GLOB LTC_BACKEND_CSRC CONFIGURE_DEPENDS
+        "ltc_backend/csrc/*.h"
+        "ltc_backend/csrc/*.cc"
+        "ltc_backend/csrc/*.cpp"
+        "ltc_backend/csrc/*/*.h"
+        "ltc_backend/csrc/*/*.cc"
+        "ltc_backend/csrc/*/*.cpp"
+        )
+add_library(example_mlir_ltc_backend SHARED ${LTC_BACKEND_CSRC})
+add_dependencies(example_mlir_ltc_backend
+        torch_mlir_ltc_backend
+        )
+target_link_libraries(example_mlir_ltc_backend
+        ${TORCH_LIBRARIES}
+        ${Python3_LIBRARIES}
+        torch_python
+        torch_mlir_ltc_backend
+        )
+
+message(STATUS "TORCH_CXXFLAGS=${TORCH_CXXFLAGS} -Wno-pedantic")
+set_target_properties(example_mlir_ltc_backend PROPERTIES
+        LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/ltc_backend/"
+        OUTPUT_NAME _EXAMPLE_MLIR_BACKEND
+        PREFIX "${PYTHON_MODULE_PREFIX}"
+        SUFFIX "${PYTHON_MODULE_EXTENSION}"
+        CXX_VISIBILITY_PRESET "hidden"
+        COMPILE_FLAGS "${TORCH_CXXFLAGS} -Wno-pedantic"
+        )
+
+###########################################################################
+# Compile Options
+###########################################################################
+
+add_compile_options(-Wno-sign-compare)
+add_compile_options(-Wno-unknown-pragmas)
+add_compile_options(-Wno-deprecated-declarations)
+add_compile_options(-Wno-return-type)
+
+if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
```

Comment:
These seem pretty close to the defaults that CMake already chooses. Can we remove these in the interest of simplicity?